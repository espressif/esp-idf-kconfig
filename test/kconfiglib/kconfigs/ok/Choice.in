# test if choice keyword is working
mainmenu "mainmenu"

    config A
        bool "a"
        default y
        help
            help for a

    comment "Named choice"

    choice NAMED_CHOICE
        prompt "prompt for named choice"

        config FOO
            bool "foo"
            help
                help for foo

        config BAR
            bool "bar"
            help
                help for bar

    endchoice

    comment "Unnamed choice"

    choice
        bool "bool choice"
        default BAZ
        help
            help for bool choice

        config BAZ
            bool "baz"
            help
                help for baz
            depends on A

        config QUX
            bool "qux"
            help
                help for qux

    endchoice
