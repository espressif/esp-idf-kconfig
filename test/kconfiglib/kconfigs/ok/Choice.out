#
# Automatically generated file. DO NOT EDIT.
# Espressif IoT Development Framework (ESP-IDF)  Project Configuration
#
CONFIG_A=y

#
# Named choice
#
CONFIG_FOO=y
# CONFIG_BAR is not set

#
# Unnamed choice
#
CONFIG_BAZ=y
# CONFIG_QUX is not set
